src/main/java/dk/kb/util/webservice/Service2ServiceRequest.java

```diff
@@ -72,10 +78,13 @@ public static <T> T httpCallWithOAuthToken (URI uri , String httpMethod, T objec
             
             String json = IOUtils.toString(con.getInputStream(), StandardCharsets.UTF_8);          
 
-            ObjectMapper mapper = new ObjectMapper();
-            @SuppressWarnings("unchecked")
-            T dto = (T) mapper.readValue(json, objectClass.getClass());           
-            return dto;
+            if (objectClass != null) { //Convert to DTO
+               ObjectMapper mapper = new ObjectMapper();
+               @SuppressWarnings("unchecked")
+               T dto = (T) mapper.readValue(json, objectClass.getClass());           
+               return dto;
```

Comment:
From the code it self it seems a bit magical and confusing why we are casting this object. Maybe extract to a method and add a simple javadoc on why we cast to T dto.

Reply:
Is little magical indeed. But required. Added comment